quantecon/discrete_var.py

```diff
@@ -0,0 +1,159 @@
+            the states that are never visited. 
+        -   S is a matrix where the element (i,j) of S is the discretized value of the j-th element of x_t in state i. Reducing S to its
+            unique values yields the grid values.
+        -   Xvec is a matrix of size m by T containing the simulated time series of the m discretized states.    
+        
+        Ancillary Functions:
+        -   Mom function: computes the VAR(1)'s unconditional variance.
+        
+        # Example:
+
+        -   This example discretizes  the sthocastic process used to calibrate the economic model included in ``Downward Nominal Wage Rigidity, 
+            Currency Pegs, and Involuntary Unemployment'' by Stephanie Schmitt-Grohé and Martín Uribe, Journal of Political Economy 
+            124, October 2016, 1466-1514. 
+                
+                A     = np.array([[0.7901, -1.3570], [-0.0104, 0.8638]])
+                omega = np.array([[0.0012346, -0.0000776], [-0.0000776, 0.0000401]])
+                N = np.array([21, 11])
+                Pii, Xvec, S = tpm(A, omega, N, T=1_000_000, Tburn = 100_000)
+
+        
+    """
+        
+    m = len(A)
+    r = len(omega)
+    
+    if np.isnan(np.sum(N)):
+        N = 10 * np.ones(m, dtype = int)
+    
+    Sigg = mom(np.eye(m), A, omega)
+    sigg = np.sqrt(np.diagonal(Sigg))  # Unconditional  standard deviation  of AR process
+    
+    
+    if np.isnan(np.sum(UB)):
+        UB = np.sqrt(10) * sigg
+    
+    V = []
+    for i in range(m):
+        b = np.linspace(-UB[i], UB[i], N[i])
+        V.append(b)
+
+    n = N.prod()     # Total number of possible values of the discretized state
+    S = np.zeros([n, m])
+    
+    for i in range(m):
+        if i == 0:
+            temp    = np.ravel(V[i])
+            S[:, i] = np.ravel(np.tile(temp,[np.prod(N[i+1:]), 1] ))
+        else:
+            temp = np.sort(np.ravel(np.tile(V[i],[np.prod(N[0:i]), 1])))
+            S[:, i] = np.ravel(np.tile(temp,[np.prod(N[i+1:]), 1] ))
+
+    Pi = np.zeros([n, n])
+    x0 = np.zeros([m, 1])
+    xx = np.zeros([n, m])
+    d = np.sum((S-xx)**2, 1)
+    ind_i = np.argmin(d)
+    Xvec = np.zeros([m, T])
+    mean = [0] * m
+    
+    for t in range(T+Tburn):
+        drw = multivariate_normal(mean, omega).reshape(1,2)
+        x =A @ x0 + drw.T
+        
+        xx = np.tile(x.T,[n,1])
+        d = np.sum((S-xx)**2, 1)
+        ind_j = np.argmin(d)
+
+        if t > Tburn:
+            Pi[ind_i,ind_j] = Pi[ind_i, ind_j] + 1
+            Xvec[:,t-Tburn] = x.T
+        
+        x0 = x
+        ind_i = ind_j
+        
+        if np.mod(t,100_000)==0:
+             print(t)
+    
+    indx = np.where(np.sum(Pi,0)>0)
+    Pi = Pi[indx[0], :]
+    Pi = Pi[:,indx[0]]
+    S  = S[indx[0], :]
+    
+    sum_row = np.sum(Pi,1)
+    for i in range(len(Pi)):
+        Pi[i,:] = Pi[i,:]/sum_row[i]
+    
+    return Pi, Xvec, S
+
+
+def mom(gx, hx, varshock, J=0, method=True):
```

Comment:
@crondonm , FYI, this code is already available in `quantecon` as `solve_discrete_lyapunov`.  I've changed it to use that function (after confirming identical results for the test matrices you use.

Reply:
Noted. Thanks for the suggestion.